travel-event-host-app/src/components/searchSection/searchEventsFilterBox/filterBox.tsx

```diff
@@ -0,0 +1,113 @@
+'use client';
+import { useEffect, useState } from 'react';
+import {
+  Box,
+  MenuItem,
+  Select,
+  Checkbox,
+  FormGroup,
+  FormControlLabel,
+  Typography,
+  IconButton,
+} from '@mui/material';
+import categoriesArr from '@/lib/categoryArray';
+import CategoriesType from '@/components/searchSection/types';
+import styles from '../styles.module.css';
+import CloseIcon from '@mui/icons-material/Close';
+
+const FilterBox = ({
+  setCategories,
+  categories,
+  filterBoxIsOpen,
+  setFilterBoxIsOpen,
+}: {
+  setCategories: React.Dispatch<React.SetStateAction<CategoriesType>>;
+  categories: CategoriesType;
+  filterBoxIsOpen: boolean;
+  setFilterBoxIsOpen: React.Dispatch<React.SetStateAction<boolean>>;
+}) => {
+  const [day, setDay] = useState('Any day');
+  const [distance, setDistance] = useState('Any distance');
+  useEffect(() => {
+    const updateCategories = () => {
+      const categoryKeys = Object.keys(categoriesArr);
+      let updateCategories: { [key: string]: boolean } = {};
+
+      Object.values(categoriesArr).map((category) => (updateCategories[category] = false));
+      setCategories(updateCategories);
+    };
+
+    updateCategories();
```

Comment:
To make this easier to read we could put `updateCategories` outside of the component (how about renaming the function to `getUpdatedCategories` and have it return the `CategoriesType` object?

Then in the useEffect you could:
```setCategories(getUpdatedCategories())```

Reply:
how I said you this updateCategories are a new type of categories this is a dynamic way to set the categories checkboxes on filterBox so I can't return CategoriesType